webapp/src/components/CollectionSelect.vue

```diff
@@ -119,23 +126,47 @@ export default {
         loading(false);
       }, debounceTime);
     },
+    IDValidationMessage() {
+      if (this.searchQuery && !/^[a-zA-Z0-9_-]+$/.test(this.searchQuery)) {
+        return "ID can only contain alphanumeric characters, dashes ('-'), and underscores ('_').";
+      }
+      if ((this.searchQuery && this.searchQuery.length < 1) || this.searchQuery.length > 40) {
+        return "ID must be between 1 and 40 characters.";
+      }
+      return "";
+    },
     async handleCreateNewCollection() {
-      try {
-        let collection_id = this.searchQuery;
-        const newCollection = await createNewCollection(this.searchQuery);
-        if (newCollection) {
-          this.value = [
-            ...this.value.filter((item) => item.collection_id !== null),
-            {
-              collection_id: collection_id,
-            },
-          ];
+      if (this.IDValidationMessage()) {
+        return;
+      } else {
+        try {
+          let collection_id = this.searchQuery;
+          const newCollection = await createNewCollection(this.searchQuery);
+          if (newCollection) {
+            this.value = [
+              ...this.value.filter((item) => item.collection_id !== null),
+              {
+                collection_id: collection_id,
+              },
+            ];
+          }
+        } catch (error) {
+          console.error("Error:", error);
+          alert("An error occurred while creating the collection. Please try again.");
```

Comment:
```suggestion
          alert("An error occurred while creating the collection. Please check that your desired collection ID is valid.");
```

Just a minor tweak; I got this message when I tried an invalid collection ID that we weren't catching, other case might be that someone else made the collection in the meantime, so "try again" probably doesn't help as they will use the same input